app/src/main/kotlin/com/simplemobiletools/calendar/pro/fragments/WeekFragment.kt

```diff
@@ -696,6 +697,12 @@ class WeekFragment : Fragment(), WeeklyCalendar {
         }
     }
 
+    private fun shouldAddEventOnTopBar(isAllDay: Boolean, startDayCode: String, endDayCode: String): Boolean {
+        val spansMultipleDays = startDayCode != endDayCode
+        val isSingleDayAllDayEvent = isAllDay && !spansMultipleDays
+        return isSingleDayAllDayEvent || spansMultipleDays && config.showMidnightSpanningEventsAtTop
```

Comment:
lets add brackets here to guarantee readable priorities

Reply:
:+1: